README.md
<!--
 * @Author: lipeng 1162423147@qq.com
 * @Date: 2023-09-24 10:46:36
 * @LastEditors: lipeng 1162423147@qq.com
 * @LastEditTime: 2023-09-24 10:47:21
 * @FilePath: /phoenix_empty/README.md
 * @Description: 这是默认设置,请设置`customMade`, 打开koroFileHeader查看配置 进行设置: https://github.com/OBKoro1/koro1FileHeader/wiki/%E9%85%8D%E7%BD%AE
-->
<!--
This README describes the package. If you publish this package to pub.dev,
this README's contents appear on the landing page for your package.

For information about how to write a good package README, see the guide for
[writing package pages](https://dart.dev/guides/libraries/writing-package-pages).

For general information about developing packages, see the Dart guide for
[creating packages](https://dart.dev/guides/libraries/create-library-packages)
and the Flutter guide for
[developing packages and plugins](https://flutter.dev/developing-packages).
-->

phoenix 将做为企业级基础组件:Empty,提供项目支持。

## Features

phoenix 将做为企业级基础组件:Empty,提供项目支持。

## Getting started

phoenix 将做为企业级基础组件:Empty,提供项目支持。

## Usage

phoenix 将做为企业级基础组件:Empty,提供项目支持。

```dart
const like = 'sample';
```

## Additional information

phoenix 将做为企业级基础组件:Empty,提供项目支持。



CHANGELOG.md
<!--
 * @Author: lipeng 1162423147@qq.com
 * @Date: 2023-09-24 10:46:36
 * @LastEditors: lipeng 1162423147@qq.com
 * @LastEditTime: 2023-10-16 15:01:36
 * @FilePath: /phoenix_empty/CHANGELOG.md
 * @Description: 这是默认设置,请设置`customMade`, 打开koroFileHeader查看配置 进行设置: https://github.com/OBKoro1/koro1FileHeader/wiki/%E9%85%8D%E7%BD%AE
-->
## 1.0.0

* [新建] 基础Empty组件

## 1.0.1

* [更新] empty_status.dart

## 1.0.2

* [更新] resource path

## 1.0.3

* [新增] example

## 1.0.4

* [新增] example

test/phoenix_empty_test.dart
import 'package:flutter_test/flutter_test.dart';

import 'package:phoenix_empty/phoenix_empty.dart';

void main() {
  test('adds one to input values', () {
    // final calculator = Calculator();
    // expect(calculator.addOne(2), 3);
    // expect(calculator.addOne(-7), -6);
    // expect(calculator.addOne(0), 1);
  });
}


example/lib/main.dart
import 'package:flutter/material.dart';

import 'abnormal_entry_page.dart';
import 'abnormal_state_example.dart';
import 'list_item.dart';

void main() {
  runApp(const MyApp());
}

class MyApp extends StatelessWidget {
  const MyApp({super.key});

  // This widget is the root of your application.
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      theme: ThemeData(
        // This is the theme of your application.
        //
        // TRY THIS: Try running your application with "flutter run". You'll see
        // the application has a blue toolbar. Then, without quitting the app,
        // try changing the seedColor in the colorScheme below to Colors.green
        // and then invoke "hot reload" (save your changes or press the "hot
        // reload" button in a Flutter-supported IDE, or press "r" if you used
        // the command line to start the app).
        //
        // Notice that the counter didn't reset back to zero; the application
        // state is not lost during the reload. To reset the state, use hot
        // restart instead.
        //
        // This works for code too, not just values: Most code changes can be
        // tested with just a hot reload.
        colorScheme: ColorScheme.fromSeed(seedColor: Colors.deepPurple),
        useMaterial3: true,
      ),
      home: AbnormalStatesEntryPage('Empty'),
    );
  }
}

class MyHomePage extends StatefulWidget {
  const MyHomePage({super.key, required this.title});

  // This widget is the home page of your application. It is stateful, meaning
  // that it has a State object (defined below) that contains fields that affect
  // how it looks.

  // This class is the configuration for the state. It holds the values (in this
  // case the title) provided by the parent (in this case the App widget) and
  // used by the build method of the State. Fields in a Widget subclass are
  // always marked "final".

  final String title;

  @override
  State<MyHomePage> createState() => _MyHomePageState();
}

class _MyHomePageState extends State<MyHomePage> {
  int _counter = 0;

  void _incrementCounter() {
    setState(() {
      // This call to setState tells the Flutter framework that something has
      // changed in this State, which causes it to rerun the build method below
      // so that the display can reflect the updated values. If we changed
      // _counter without calling setState(), then the build method would not be
      // called again, and so nothing would appear to happen.
      _counter++;
    });
  }

  @override
  Widget build(BuildContext context) {
    // This method is rerun every time setState is called, for instance as done
    // by the _incrementCounter method above.
    //
    // The Flutter framework has been optimized to make rerunning build methods
    // fast, so that you can just rebuild anything that needs updating rather
    // than having to individually change instances of widgets.
    return Scaffold(
      appBar: AppBar(
        // TRY THIS: Try changing the color here to a specific color (to
        // Colors.amber, perhaps?) and trigger a hot reload to see the AppBar
        // change color while the other colors stay the same.
        backgroundColor: Theme.of(context).colorScheme.inversePrimary,
        // Here we take the value from the MyHomePage object that was created by
        // the App.build method, and use it to set our appbar title.
        title: Text(widget.title),
      ),
      body: ListView(
        children: <Widget>[
          ListItem(
            title: "异常信息+操作",
            isSupportTheme: true,
            isShowLine: false,
            describe: '异常信息+操作',
            onPressed: () {
              Navigator.push(context, MaterialPageRoute(
                builder: (BuildContext context) {
                  return AbnomalStateExample(
                    caseIndex: 0,
                  );
                },
              ));
            },
          ),
          ListItem(
            title: "异常信息居中展示",
            isSupportTheme: true,
            describe: '异常信息居中展示',
            onPressed: () {
              Navigator.push(context, MaterialPageRoute(
                builder: (BuildContext context) {
                  return AbnomalStateExample(
                    caseIndex: 1,
                  );
                },
              ));
            },
          ),
          ListItem(
            title: "异常信息默认展示",
            isSupportTheme: true,
            describe: '异常信息默认展示',
            onPressed: () {
              Navigator.push(context, MaterialPageRoute(
                builder: (BuildContext context) {
                  return AbnomalStateExample(
                    caseIndex: 2,
                  );
                },
              ));
            },
          ),
          ListItem(
            title: "大模块空态",
            isSupportTheme: true,
            describe: '大模块空态',
            onPressed: () {
              Navigator.push(context, MaterialPageRoute(
                builder: (BuildContext context) {
                  return AbnomalStateExample(
                    caseIndex: 3,
                  );
                },
              ));
            },
          ),
          ListItem(
            title: "单按钮效果",
            isSupportTheme: true,
            describe: '单按钮效果',
            onPressed: () {
              Navigator.push(context, MaterialPageRoute(
                builder: (BuildContext context) {
                  return AbnomalStateExample(
                    caseIndex: 4,
                  );
                },
              ));
            },
          ),
          ListItem(
            title: "双按钮效果",
            isSupportTheme: true,
            describe: '双按钮效果',
            onPressed: () {
              Navigator.push(context, MaterialPageRoute(
                builder: (BuildContext context) {
                  return AbnomalStateExample(
                    caseIndex: 5,
                  );
                },
              ));
            },
          ),
          ListItem(
            title: "小模块空态",
            isSupportTheme: true,
            describe: '小模块空态',
            onPressed: () {
              Navigator.push(context, MaterialPageRoute(
                builder: (BuildContext context) {
                  return AbnomalStateExample(
                    caseIndex: 6,
                  );
                },
              ));
            },
          ),
        ],
      ),
      floatingActionButton: FloatingActionButton(
        onPressed: _incrementCounter,
        tooltip: 'Increment',
        child: const Icon(Icons.add),
      ), // This trailing comma makes auto-formatting nicer for build methods.
    );
  }
}


lib/empty_status.dart
import 'package:flutter/material.dart';
import 'package:phoenix_base/phoenix.dart';
import 'package:phoenix_empty/extension/empty_total_config.dart';

import 'config/abnormal_state_config.dart';
import 'extension/empty_assets.dart';

/// 页面状态
enum AbnormalState {
  /// 获取数据异常
  getDataFailed,

  /// 网络连接异常
  networkConnectError,

  /// 暂无数据
  noData
}

/// /// /// /// /// /// /// /// /// /
/// 描述: 异常页面展示
/// 注：这里仅提供几种常用异常页面，需要其他内容异常页面自行实现 EmptyStateWidget
/// /// /// /// /// /// /// /// /// /
class AbnormalStateUtils {
  /// 通过状态获取对应空页面widget
  /// status: 页面状态类型为[EmptyState]
  static Widget getEmptyWidgetByState(
      BuildContext context, AbnormalState status,
      {Image? img, EmptyStatusIndexedActionClickCallback? action}) {
    if (AbnormalState.getDataFailed == status) {
      return AbnormalStateWidget(
        img: img ?? PhoenixTools.getAssetImage(EmptyAssets.noData),
        title: BrnIntl.of(context).localizedResource.fetchErrorAndRetry,
        operateTexts: <String>[
          BrnIntl.of(context).localizedResource.clickPageAndRetry
        ],
        action: action,
      );
    } else if (AbnormalState.networkConnectError == status) {
      return AbnormalStateWidget(
        img: img ?? PhoenixTools.getAssetImage(EmptyAssets.networkError),
        title: BrnIntl.of(context).localizedResource.netErrorAndRetryLater,
        operateTexts: <String>[
          BrnIntl.of(context).localizedResource.clickPageAndRetry
        ],
        action: action,
      );
    } else if (AbnormalState.noData == status) {
      return AbnormalStateWidget(
          img: img ?? PhoenixTools.getAssetImage(EmptyAssets.noData),
          title: BrnIntl.of(context).localizedResource.noDataTip);
    } else {
      return const SizedBox.shrink();
    }
  }
}

/// 操作区域按钮类型
enum OperateAreaType {
  /// 单按钮
  singleButton,

  /// 双按钮
  doubleButton,

  /// 文本按钮
  textButton
}

/// 空页面操作区域按钮的点击回调
/// index: 被点击按钮的索引
typedef EmptyStatusIndexedActionClickCallback = void Function(int index);

/// 异常页面展示一般用于网络错误、数据为空的提示和引导
// ignore: must_be_immutable
class AbnormalStateWidget extends StatelessWidget {
  /// 图片
  final Image? img;

  /// 标题
  final String? title;

  /// 内容
  final String? content;

  /// 操作区类型
  final OperateAreaType operateAreaType;

  /// 操作区文案
  final List<String>? operateTexts;

  /// 点击事件回调
  final EmptyStatusIndexedActionClickCallback? action;

  /// 是否可点击页面回调配合[action]使用
  /// 当为true时调用[action]回调，当为false时不做处理
  /// 默认false
  final bool enablePageTap;

  /// 顶部距离走自动计算逻辑：父视图高度的8%，可自己指定高度
  /// 默认为null
  final double? topOffset;

  /// 背景色设置
  /// 默认Colors.white
  final Color bgColor;

  /// 距顶部高度百分比
  final double topPercent;

  /// 内容垂直居中
  /// 默认 false
  final bool isCenterVertical;

  AbnormalStateConfig? themeData;

  AbnormalStateWidget({
    super.key,
    this.img,
    this.title,
    this.content,
    this.operateAreaType = OperateAreaType.textButton,
    this.operateTexts,
    this.action,
    this.enablePageTap = false,
    this.topOffset,
    this.bgColor = Colors.white,
    this.isCenterVertical = false,
    this.topPercent = 0.08,
    this.themeData,
  }) {
    themeData ??= AbnormalStateConfig();
    themeData = BaseThemeConfig.instance
        .getConfig(configId: themeData!.configId)
        .abnormalStateConfig
        .merge(themeData);
  }

  @override
  Widget build(BuildContext context) {
    return GestureDetector(
        onTap: () {
          if (enablePageTap && action != null) {
            action!(0);
          }
        },
        child: Container(
          color: bgColor,
          child: Column(
            mainAxisSize: MainAxisSize.max,
            mainAxisAlignment: isCenterVertical
                ? MainAxisAlignment.center
                : MainAxisAlignment.start,
            children: <Widget>[
              _buildImageWidget(context),
              _buildTextWidget(),
              _buildContentWidget(),
              _buildOperateWidget(),
            ],
          ),
        ));
  }

  ///图片区域
  ///要求顶部距离是父布局的8%
  _buildImageWidget(BuildContext context) {
    final size = MediaQuery.of(context).size;
    final height = size.height;
    return img != null
        ? Container(
            padding: isCenterVertical
                ? null
                : EdgeInsets.only(top: topOffset ?? height * topPercent),
            child: img,
          )
        : const SizedBox.shrink();
  }

  ///文案区域：标题
  _buildTextWidget() {
    return title != null
        ? Container(
            alignment: Alignment.center,
            padding: const EdgeInsets.fromLTRB(60, 24, 60, 0),
            child: Text(title!,
                textAlign: TextAlign.center,
                style: themeData!.titleTextStyle.generateTextStyle()),
          )
        : const SizedBox.shrink();
  }

  ///文案区域：内容
  _buildContentWidget() {
    return content != null
        ? Container(
            alignment: Alignment.center,
            padding: const EdgeInsets.fromLTRB(60, 12, 60, 0),
            child: Text(content!,
                textAlign: TextAlign.center,
                style: themeData!.contentTextStyle.generateTextStyle()),
          )
        : const SizedBox.shrink();
  }

  ///操作区域
  _buildOperateWidget() {
    return operateTexts != null
        ? Container(
            padding: const EdgeInsets.only(top: 36),
            child: _buildOperateContentWidget(),
          )
        : const SizedBox.shrink();
  }

  ///操作区按钮
  _buildOperateContentWidget() {
    if (OperateAreaType.singleButton == operateAreaType) {
      return GestureDetector(
        onTap: () {
          if (action != null) action!(0);
        },
        child: Container(
          constraints: BoxConstraints(minWidth: themeData!.singleMinWidth),
          padding: const EdgeInsets.fromLTRB(48, 16, 48, 16),
          decoration: BoxDecoration(
              color: themeData!.commonConfig.brandPrimary,
              borderRadius:
                  BorderRadius.all(Radius.circular(themeData!.btnRadius))),
          child: Text(operateTexts![0],
              textAlign: TextAlign.center,
              style: themeData!.singleTextStyle.generateTextStyle()),
        ),
      );
    } else if (OperateAreaType.doubleButton == operateAreaType) {
      return Row(
        mainAxisAlignment: MainAxisAlignment.center,
        children: <Widget>[
          GestureDetector(
            onTap: () {
              if (action != null) action!(0);
            },
            child: Container(
              constraints: BoxConstraints(minWidth: themeData!.doubleMinWidth),
              padding: const EdgeInsets.fromLTRB(36, 16, 36, 16),
              decoration: BoxDecoration(
                  color: themeData!.commonConfig.brandPrimary.withAlpha(0x14),
                  borderRadius:
                      BorderRadius.all(Radius.circular(themeData!.btnRadius))),
              child: Text(operateTexts![0],
                  textAlign: TextAlign.center,
                  style: themeData!.doubleTextStyle.generateTextStyle()),
            ),
          ),
          Container(
            width: 12,
            color: Colors.transparent,
          ),
          GestureDetector(
            onTap: () {
              if (action != null) action!(1);
            },
            child: Container(
              constraints: BoxConstraints(minWidth: themeData!.doubleMinWidth),
              padding: const EdgeInsets.fromLTRB(36, 16, 36, 16),
              decoration: BoxDecoration(
                  color: themeData!.commonConfig.brandPrimary.withAlpha(0x14),
                  borderRadius:
                      BorderRadius.all(Radius.circular(themeData!.btnRadius))),
              child: Text(operateTexts![1],
                  textAlign: TextAlign.center,
                  style: themeData!.doubleTextStyle.generateTextStyle()),
            ),
          ),
        ],
      );
    } else if (OperateAreaType.textButton == operateAreaType) {
      return GestureDetector(
          onTap: () {
            if (action != null) action!(0);
          },
          child: Text(operateTexts![0],
              style: themeData!.operateTextStyle.generateTextStyle()));
    }
    return const SizedBox.shrink();
  }
}


lib/config/abnormal_state_config.dart
import 'package:phoenix_base/phoenix.dart';
import 'package:phoenix_empty/extension/empty_total_config.dart';

import '../extension/empty_default_config_utils.dart';

/// 描述: 空页面配置类
class AbnormalStateConfig extends BaseConfig {
  AbnormalStateConfig({
    BaseTextStyle? titleTextStyle,
    BaseTextStyle? contentTextStyle,
    BaseTextStyle? operateTextStyle,
    double? btnRadius,
    BaseTextStyle? singleTextStyle,
    BaseTextStyle? doubleTextStyle,
    double? singleMinWidth,
    double? doubleMinWidth,
    String configId = GLOBAL_CONFIG_ID,
  })  : _titleTextStyle = titleTextStyle,
        _contentTextStyle = contentTextStyle,
        _operateTextStyle = operateTextStyle,
        _btnRadius = btnRadius,
        _singleTextStyle = singleTextStyle,
        _doubleTextStyle = doubleTextStyle,
        _singleMinWidth = singleMinWidth,
        _doubleMinWidth = doubleMinWidth,
        super(configId: configId);

  /// 文案区域标题
  ///
  /// BaseTextStyle(
  ///   color: [BrnCommonConfig.colorTextBase],
  ///   fontSize: [BrnCommonConfig.fontSizeSubHead],
  ///   fontWeight: FontWeight.w600,
  /// )
  BaseTextStyle? _titleTextStyle;

  /// 文案区域内容
  ///
  /// BaseTextStyle(
  ///   color: [BrnCommonConfig.colorTextHint],
  ///   fontSize: [BrnCommonConfig.fontSizeBase],
  /// )
  BaseTextStyle? _contentTextStyle;

  /// 操作区域文本样式
  ///
  /// BaseTextStyle(
  ///   color: [BrnCommonConfig.brandPrimary],
  ///   fontSize: [BrnCommonConfig.fontSizeBase],
  /// )
  BaseTextStyle? _operateTextStyle;

  /// 圆角
  /// default value is [BrnCommonConfig.radiusSm]
  double? _btnRadius;

  /// 单按钮文本样式
  ///
  /// BaseTextStyle(
  ///   color: [BrnCommonConfig.colorTextBaseInverse],
  ///   fontSize: [BrnCommonConfig.fontSizeSubHead],
  /// )
  BaseTextStyle? _singleTextStyle;

  /// 双按钮文本样式
  ///
  /// BaseTextStyle(
  ///   color: [BrnCommonConfig.brandPrimary],
  ///   fontSize: [BrnCommonConfig.fontSizeSubHead],
  /// )
  BaseTextStyle? _doubleTextStyle;

  /// 单按钮的按钮最小宽度
  /// 默认值为 160
  double? _singleMinWidth;

  /// 多按钮的按钮最小宽度
  /// 默认值为 120
  double? _doubleMinWidth;

  BaseTextStyle get titleTextStyle =>
      _titleTextStyle ??
      BaseEmptyConfigUtils.defaultAbnormalStateConfig.titleTextStyle;

  BaseTextStyle get contentTextStyle =>
      _contentTextStyle ??
      BaseEmptyConfigUtils.defaultAbnormalStateConfig.contentTextStyle;

  BaseTextStyle get operateTextStyle =>
      _operateTextStyle ??
      BaseEmptyConfigUtils.defaultAbnormalStateConfig.operateTextStyle;

  double get btnRadius =>
      _btnRadius ?? BaseEmptyConfigUtils.defaultAbnormalStateConfig.btnRadius;

  BaseTextStyle get singleTextStyle =>
      _singleTextStyle ??
      BaseEmptyConfigUtils.defaultAbnormalStateConfig.singleTextStyle;

  BaseTextStyle get doubleTextStyle =>
      _doubleTextStyle ??
      BaseEmptyConfigUtils.defaultAbnormalStateConfig.doubleTextStyle;

  double get singleMinWidth =>
      _singleMinWidth ??
      BaseEmptyConfigUtils.defaultAbnormalStateConfig.singleMinWidth;

  double get doubleMinWidth =>
      _doubleMinWidth ??
      BaseEmptyConfigUtils.defaultAbnormalStateConfig.doubleMinWidth;

  @override
  void initThemeConfig(
    String configId, {
    BaseCommonConfig? currentLevelCommonConfig,
  }) {
    super.initThemeConfig(
      configId,
      currentLevelCommonConfig: currentLevelCommonConfig,
    );

    /// 用户全局组件配置
    AbnormalStateConfig abnormalStateConfig = BaseThemeConfig.instance
        .getConfig(configId: configId)
        .abnormalStateConfig;

    _titleTextStyle = abnormalStateConfig.titleTextStyle.merge(
      BaseTextStyle(
        color: commonConfig.colorTextBase,
        fontSize: commonConfig.fontSizeSubHead,
      ).merge(_titleTextStyle),
    );
    _contentTextStyle = abnormalStateConfig.contentTextStyle.merge(
      BaseTextStyle(
        color: commonConfig.colorTextHint,
        fontSize: commonConfig.fontSizeSubHead,
      ).merge(_contentTextStyle),
    );
    _operateTextStyle = abnormalStateConfig.operateTextStyle.merge(
      BaseTextStyle(
        color: commonConfig.brandPrimary,
        fontSize: commonConfig.fontSizeBase,
      ).merge(_operateTextStyle),
    );
    _singleTextStyle = abnormalStateConfig.singleTextStyle.merge(
      BaseTextStyle(
        color: commonConfig.colorTextBaseInverse,
        fontSize: commonConfig.fontSizeSubHead,
      ).merge(_singleTextStyle),
    );
    _doubleTextStyle = abnormalStateConfig.doubleTextStyle.merge(
      BaseTextStyle(
        color: commonConfig.brandPrimary,
        fontSize: commonConfig.fontSizeSubHead,
      ).merge(_doubleTextStyle),
    );
    _btnRadius ??= abnormalStateConfig._btnRadius;
    _singleMinWidth ??= abnormalStateConfig._singleMinWidth;
    _doubleMinWidth ??= abnormalStateConfig._doubleMinWidth;
  }

  AbnormalStateConfig copyWith({
    BaseTextStyle? titleTextStyle,
    BaseTextStyle? contentTextStyle,
    BaseTextStyle? operateTextStyle,
    double? btnRadius,
    BaseTextStyle? singleTextStyle,
    BaseTextStyle? doubleTextStyle,
    double? singleMinWidth,
    double? doubleMinWidth,
  }) {
    return AbnormalStateConfig(
      titleTextStyle: titleTextStyle ?? _titleTextStyle,
      contentTextStyle: contentTextStyle ?? _contentTextStyle,
      operateTextStyle: operateTextStyle ?? _operateTextStyle,
      btnRadius: btnRadius ?? _btnRadius,
      singleTextStyle: singleTextStyle ?? _singleTextStyle,
      doubleTextStyle: doubleTextStyle ?? _doubleTextStyle,
      singleMinWidth: singleMinWidth ?? _singleMinWidth,
      doubleMinWidth: doubleMinWidth ?? _doubleMinWidth,
    );
  }

  AbnormalStateConfig merge(AbnormalStateConfig? other) {
    if (other == null) return this;
    return copyWith(
      titleTextStyle: titleTextStyle.merge(other._titleTextStyle),
      contentTextStyle: contentTextStyle.merge(other._contentTextStyle),
      operateTextStyle: operateTextStyle.merge(other._operateTextStyle),
      btnRadius: other._btnRadius,
      singleTextStyle: singleTextStyle.merge(other._singleTextStyle),
      doubleTextStyle: doubleTextStyle.merge(other._doubleTextStyle),
      singleMinWidth: other._singleMinWidth,
      doubleMinWidth: other._doubleMinWidth,
    );
  }
}


lib/extension/empty_total_config.dart
/*
 * @Author: lipeng 1162423147@qq.com
 * @Date: 2023-09-24 11:00:29
 * @LastEditors: lipeng 1162423147@qq.com
 * @LastEditTime: 2023-10-11 19:20:05
 * @FilePath: /phoenix_empty/lib/extension/empty_total_config.dart
 * @Description: 这是默认设置,请设置`customMade`, 打开koroFileHeader查看配置 进行设置: https://github.com/OBKoro1/koro1FileHeader/wiki/%E9%85%8D%E7%BD%AE
 */
import 'package:phoenix_base/phoenix.dart';

import '../config/abnormal_state_config.dart';
import 'empty_default_config_utils.dart';

class EmptyTotalConfig extends BaseTotalConfig {
  EmptyTotalConfig({AbnormalStateConfig? emptyConfig})
      : _emptyConfig = emptyConfig;

  AbnormalStateConfig? _emptyConfig;

  AbnormalStateConfig get emptyConfig =>
      _emptyConfig ?? BaseEmptyConfigUtils.defaultAbnormalStateConfig;

  @override
  void initThemeConfig(String configId) {
    super.initThemeConfig(configId);
    _emptyConfig ??= AbnormalStateConfig();
    emptyConfig.initThemeConfig(
      configId,
      currentLevelCommonConfig: commonConfig,
    );
  }
}

extension BaseEmptyTotalConfig on BaseTotalConfig {
  static AbnormalStateConfig? _emptyConfig;
  AbnormalStateConfig get abnormalStateConfig =>
      _emptyConfig ?? BaseEmptyConfigUtils.defaultAbnormalStateConfig;
  set abnormalStateTotalConfig(EmptyTotalConfig config) {
    _emptyConfig = config.emptyConfig;
  }
}


example/lib/abnormal_state_example.dart
import 'package:flutter/material.dart';
import 'package:phoenix_base/phoenix.dart';
import 'package:phoenix_empty/phoenix_empty.dart';

class AbnomalStateExample extends StatelessWidget {
  final int? caseIndex;

  AbnomalStateExample({this.caseIndex});

  @override
  Widget build(BuildContext context) {
    return Scaffold(
        appBar: AppBar(
          title: Text('异常页面'),
        ),
        body: _buildEmpty(caseIndex, context));
  }

  Widget? _buildEmpty(int? index, BuildContext context) {
    Widget? widget;
    switch (index) {
      case 0:
        widget = AbnormalStateWidget(
          img: Image.asset(
            'assets/image/content_failed.png',
            scale: 3.0,
          ),
          isCenterVertical: true,
          title: "获取数据失败，请重试",
          operateTexts: <String>["请点击页面重试"],
          operateAreaType: OperateAreaType.textButton,
          action: (index) {
            //BrnToast.show("获取数据失败，请重试", context);
          },
        );
        break;
      case 1:
        widget = AbnormalStateWidget(
          isCenterVertical: true,
          img: Image.asset(
            'assets/image/no_data.png',
            scale: 3.0,
          ),
          title: BrnIntl.of(context).localizedResource.noDataTip,
        );
        break;
      case 2:
        widget = AbnormalStateWidget(
          img: Image.asset(
            'assets/image/network_error.png',
            scale: 3.0,
          ),
          title: '网络数据异常',
        );
        break;
      case 3:
        widget = AbnormalStateWidget(
          img: Image.asset(
            'assets/image/no_data.png',
            scale: 3.0,
          ),
          content: '您的门店暂无用户',
        );
        break;
      case 4:
        widget = AbnormalStateWidget(
          img: Image.asset(
            'assets/image/no_data.png',
            scale: 3.0,
          ),
          title: "这是副标题内容这是副标题内容这是副标",
          content: '您的门店暂无用户',
          operateAreaType: OperateAreaType.singleButton,
          operateTexts: ["切换账号"],
          action: (_) {
            //BrnToast.show("第$_个按钮被点击了", context);
          },
        );
        break;
      case 5:
        widget = AbnormalStateWidget(
          img: Image.asset(
            'assets/image/no_data.png',
            scale: 3.0,
          ),
          title: "暂无",
          content: '您还没有在维护的信息哦',
          operateAreaType: OperateAreaType.doubleButton,
          operateTexts: ['去添加', '去修改'],
          action: (_) {
            // BrnToast.show("第$_个按钮被点击了", context);
          },
        );
        break;
      case 6:
        widget = AbnormalStateWidget(
          content: '您的门店暂无用户',
        );
        break;
    }

    return widget;
  }
}


lib/extension/empty_default_config_utils.dart
import 'package:flutter/widgets.dart';
import 'package:phoenix_base/phoenix.dart';

import '../config/abnormal_state_config.dart';

/*
 * @Author: lipeng 1162423147@qq.com
 * @Date: 2023-09-24 10:52:35
 * @LastEditors: lipeng 1162423147@qq.com
 * @LastEditTime: 2023-09-24 10:55:04
 * @FilePath: /phoenix_empty/lib/extension/empty_default_config_utils.dart
 * @Description: 这是默认设置,请设置`customMade`, 打开koroFileHeader查看配置 进行设置: https://github.com/OBKoro1/koro1FileHeader/wiki/%E9%85%8D%E7%BD%AE
 */
/// 空页面配置
extension BaseEmptyConfigUtils on BaseDefaultConfigUtils {
  //
  static AbnormalStateConfig defaultAbnormalStateConfig = AbnormalStateConfig(
    titleTextStyle: BaseTextStyle(
      color: BaseDefaultConfigUtils.defaultCommonConfig.colorTextBase,
      fontSize: BaseDefaultConfigUtils.defaultCommonConfig.fontSizeSubHead,
      fontWeight: FontWeight.w600,
    ),
    contentTextStyle: BaseTextStyle(
      color: BaseDefaultConfigUtils.defaultCommonConfig.colorTextHint,
      fontSize: BaseDefaultConfigUtils.defaultCommonConfig.fontSizeBase,
    ),
    operateTextStyle: BaseTextStyle(
      color: BaseDefaultConfigUtils.defaultCommonConfig.brandPrimary,
      fontSize: BaseDefaultConfigUtils.defaultCommonConfig.fontSizeBase,
    ),
    btnRadius: 4,
    doubleTextStyle: BaseTextStyle(
      color: BaseDefaultConfigUtils.defaultCommonConfig.brandPrimary,
      fontSize: BaseDefaultConfigUtils.defaultCommonConfig.fontSizeSubHead,
    ),
    singleTextStyle: BaseTextStyle(
      color: BaseDefaultConfigUtils.defaultCommonConfig.colorTextBaseInverse,
      fontSize: BaseDefaultConfigUtils.defaultCommonConfig.fontSizeSubHead,
    ),
    singleMinWidth: 160.0,
    doubleMinWidth: 120.0,
  );
}
